rfcs/text/0008_pulse.md

```diff
@@ -0,0 +1,310 @@
+
+I'd appreciate some insights here to come up with a strong handshake mechanism to avoid stealing identities.
+
+In order to _dereference_ the data, we can store these mappings in a Vault or Secrets provider instead of an index in our ES.
+
+_NB: Not for phase 1_
+
+##### Opt-In|Out
+
+Similar to the current telemetry, we want to keep track of when the user opts in or out of telemetry. The implementation can be very similar to the current one. But we recently learned we need to add the origin to know what application has telemetry disabled (Kibana, Beats, Enterprise Search, ...). This makes me wonder if we will ever want to provide a granular option for the user to be able to cherry-pick about what channels are sent and which ones should be disabled.
+
+##### Inject telemetry
+
+In order to minimise the amount of requests, this `POST` should accept bulks of data in the payload (mind the payload size limits if any). It will require authentication based on the `deploymentID` and `token` explained in the [previous endpoint](#authenticate) (_NB: Not for phase 1_).
+
+The received payload will be pushed to a streaming technology (AWS Firehose, Google Pub/Sub, ...). This way we can maintain a buffer in cases the ingestion of data spikes or we need to stop our ES cluster for any maintenance purposes.
+
+A subscriber to that stream will receive that info a split the payload into smaller documents per channel and index them into their separate indices.
+
+This indexing should also trigger some additional processes like the **generation of instructions** and _special views_ (only if needed, check the point [Access control](#access-control) for more details).
+
+_NB: We might want to consider some sort of piggy-backing to include the instructions in the response. But for the purpose of this RFC, scalability and separation of concerns, I'd rather keep it for future possible improvements._
+
+##### Retrieve instructions
+
+_NB: Only after phase 3_
+
+This `GET` endpoint should return the list of instructions generated for that deployment. To control the likely ever-growing list of instructions for each deployment, it will accept a `since` query parameter where the requester can specify the timestamp ever since it was to retrieve the new values.
+
+This endpoint will read the `instructions-*` indices, filtering `updated-at` by the `since` query parameter (if provided) and it will return the results, grouping them by channels.
+
+Additionally, we can consider accepting an additional query parameter to retrieve only specific channels. For use cases like distributed components (endpoint, apm, beats, ...) polling instructions themselves.
+
+#### Data model
+
+The storage of each of the documents, will be based on monthly-rolling indices split by channels. This means we'll have indices like `pulse-raw-{CHANNEL_NAME}-YYYY.MM` and `pulse-instructions-{CHANNEL_NAME}-YYYY.MM` (final names TBD).
```

Comment:
Initially, we are thinking of system (dot) indices instead of Saved Objects. This is internally handled by the `sendToChannel` method so it should be transparent to the plugins.

Regarding the time-based indices, for time-based data (such as this new granular telemetry), our recommendations (on trainings) are to always use time-based indices for time-based data.
Do you think we should use _static_ indices instead?